packages/nimble-components/src/rich-text-editor/template.ts

```diff
@@ -1,6 +1,53 @@
-import { html } from '@microsoft/fast-element';
+import { html, ref, repeat, children, elements } from '@microsoft/fast-element';
 import type { RichTextEditor } from '.';
+import { toolbarTag } from '../toolbar';
+import { toggleButtonTag } from '../toggle-button';
+import type { EditorButton } from './models/editor-button';
 
 export const template = html<RichTextEditor>`
-    <template> Rich Text Editor here. </template>
+    <template>
+        <div class="container">
+            <section
+                ${ref('editor')}
+                class="editor"
+                role="textbox"
+                aria-multiline="true">
+            </section>
+            <section class="footer-section">
+                <${toolbarTag}
+                    ${children({
+        property: 'toggleButtons',
+        filter: elements(toggleButtonTag)
+    })}>
+                    ${repeat(
+        x => x.editorButtons,
```

Comment:
Hi @vivinkrishna-ni, I am concerned that too much abstraction is being invented here for the value that it provides.

The EditorButton class is inventing new representations for defining classes, slotted content, binding to event listeners, etc. Those are responsibilities that should be handled in a template.

If there was an argument for needing that kind of abstraction then I would not expect chunks of template to be represented as an object with properties:

```ts
       {
            class: 'bold',
            iconLabel: 'Bold',
            tiptapNodeOrMarkName: 'bold',
            iconTemplate: html`<${iconBoldBTag} slot="start"></${iconBoldBTag}>`
        }
```

But as a piece of template:

```ts
const boldTemplate = html`
    <${toggleButtonTag}
        class="bold"
        @click=${x => x.click();}>
        bold
        <${iconBoldBTag} slot="start"></${iconBoldBTag}>
    </${toggleButtonTag}>
`;
```

However, in this scenario even that level of abstraction seems unnecessary. We aren't building a plugin system, etc, for the toolbar options available. The features are defined by the rich-text-editor. I would expect the buttons to just be expressed inline in the template, such as by doing:

```html
<section class="footer-section">
    <${toolbarTag}>
        <${toggleButtonTag}
            $(ref('boldButton'))
            class="bold"
            @click="${x => x.boldButtonClick()}">
        </${toggleButtonTag}>
        <!-- Other toolbar buttons each inline -->
    </${toolbarTag}>
</section>
```

This is a pretty big shift from the current implementation. If you have questions I'd encourage you to schedule a meeting for tomorrow so we can resolve concerns quickly.

Reply:
Yes, we agree with your concern. I have updated it as per your suggestion.